crates/aptos-ledger/src/lib.rs

```diff
@@ -0,0 +1,309 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! # aptos-ledger
+//!
+//! `aptos-ledger` provides convenience methods to communicate with the Aptos app on ledger
+
+#![deny(missing_docs)]
+
+use hex::encode;
+use ledger_apdu::APDUCommand;
+use ledger_transport_hid::{hidapi::HidApi, LedgerHIDError, TransportNativeHID};
+use once_cell::sync::Lazy;
+use std::{
+    fmt,
+    fmt::{Debug, Display},
+    str,
+};
+use thiserror::Error;
+
+// A piece of data which tells a wallet how to derive a specific key within a tree of keys
+// 637 is the key for Aptos
+// TODO: Add support for multiple index
+const DERIVATIVE_PATH: &str = "m/44'/637'/0'/0'/0'";
+
+const CLA_APTOS: u8 = 0x5B; // Aptos CLA Instruction class
+const INS_GET_VERSION: u8 = 0x03; // Get version instruction code
+const INS_GET_APP_NAME: u8 = 0x04; // Get app name instruction code
+const INS_GET_PUB_KEY: u8 = 0x05; // Get public key instruction code
+const INS_SIGN_TXN: u8 = 0x06; // Sign the transaction
+const APDU_CODE_SUCCESS: u16 = 36864; // Success code for transport.exchange
+
+const MAX_APDU_LEN: usize = 255;
+const P1_NON_CONFIRM: u8 = 0x00;
+const P1_CONFIRM: u8 = 0x01;
+const P1_START: u8 = 0x00;
+const P2_MORE: u8 = 0x80;
+const P2_LAST: u8 = 0x00;
+
+static SERIALIZED_BIP32: Lazy<Vec<u8>> = Lazy::new(|| serialize_bip32(DERIVATIVE_PATH));
+
+#[derive(Debug, Error)]
+/// Aptos Ledger Error
+pub enum AptosLedgerError {
+    /// Error when trying to open a connection to the Ledger device
+    #[error("Device not found")]
+    DeviceNotFound,
+
+    /// Unexpected error
+    #[error("Unexpected Error: {0}")]
+    UnexpectedError(String, Option<u16>),
```

Comment:
Could you leave a comment explaining what the Option<u16> is here?

Reply:
Also you should probably show the error code, something like this
```suggestion
    /// Unexpected error
    #[error("Unexpected Error: {0} (code {1:?})")]
    UnexpectedError(String, Option<u16>),
```